src/rule-computer.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+var Rule = require('./rule.js');
+
+module.exports = class RuleComputer {
+  /**
+   * @param config
+   * @param {IssueDataProvider} issueDataProvider
+   * @param {Logger} logger
+   */
+  constructor (config, issueDataProvider, logger) {
+    this.config = config;
+    this.logger = logger;
+    this.issueDataProvider = issueDataProvider;
+  }
+
+  /**
+   * Try to find whether webhook context matches a rule requirements.
+   *
+   * @param {Context} context
+   *
+   * @returns {Promise<*>}
+   */
+  async findRule (context) {
+
+    switch (context.name) {
+      case 'issues':
+        this.logger.debug('Rule debug: is issue');
+        var promise = this.findIssueRule(context);
+
+        return promise;
+        break;
+
+      case 'issue_comment':
+        this.logger.debug('Rule debug: is issue comment');
+        break;
+
+      default:
+        this.logger.debug('No rule applies to ' + context.name);
+    }
+  }
+
+  /**
+   * Try to find whether webhook context matches an Issue rule requirements.
+   *
+   * @param {Context} context
+   *
+   * @returns {Promise<*>}
+   *
+   * @private
+   */
+  async findIssueRule (context) {
+
+    if (context.payload.action == 'milestoned') {
+      this.logger.debug('Rule debug: milestoned');
+
+      const milestone = context.payload.issue.milestone.title;
+
+      if ((milestone == this.config.milestones.next_minor_milestone) || (milestone == this.config.milestones.next_patch_milestone)) {
+        const issueId = context.payload.issue.number;
+
```

Comment:
Empty blank line. Better to group variables definition:
https://github.com/airbnb/javascript#variables--const-let-group